apps/web/app/features/stacking/start-pooled-stacking.tsx

```diff
@@ -0,0 +1,164 @@
+
+  if (!stxAddress || !client) {
+    return 'You should connect STX wallet';
+  }
+  if (!client) {
+    return 'Expected client to be defined';
+  }
+
+  return (
+    <StartPooledStackingLayout client={client} networkName={networkName} poolName={poolName} />
+  );
+}
+
+const initialStackingFormValues: Partial<StackingFormValues> = {
+  amount: '',
+  numberOfCycles: 1,
+  poolAddress: '',
+};
+
+interface StartPooledStackingLayoutProps {
+  client: StackingClient;
+  networkName: StacksNetworkName;
+  poolName: PoolName;
+}
+
+function StartPooledStackingLayout({ client }: StartPooledStackingLayoutProps) {
+  const { stxAddress, btcAddressP2wpkh } = useLeatherConnect();
+  const { network, networkInstance } = useStacksNetwork();
+  const poxContracts = getPoxContracts(network);
+
+  const getSecondsUntilNextCycleQuery = useQuery(
+    createGetSecondsUntilNextCycleQueryOptions({ client })
+  );
+
+  const [contractAddress, contractName] = getPoxContracts(network)['Pox4'].split('.');
+
+  const getAllowanceContractCallersFastPoolQuery = useQuery(
+    createGetAllowanceContractCallersQueryOptions({
+      contractAddress,
+      contractName,
+      callingContract: poxContracts['WrapperFastPool'],
+      senderAddress: stxAddress ? stxAddress.address : null,
+      network,
+    })
+  );
+
+  const getAllowanceContractCallersRestakeQuery = useQuery(
+    createGetAllowanceContractCallersQueryOptions({
+      contractAddress,
+      contractName,
+      callingContract: poxContracts['WrapperRestake'],
+      senderAddress: stxAddress ? stxAddress.address : null,
+      network,
+    })
+  );
+
+  const getAllowanceContractCallersOneCycleQuery = useQuery(
+    createGetAllowanceContractCallersQueryOptions({
+      contractAddress,
+      contractName,
+      callingContract: poxContracts['WrapperOneCycle'],
+      senderAddress: stxAddress ? stxAddress.address : null,
+      network,
+    })
+  );
+
+  const [hasUserConfirmedPoolWrapperContract, setHasUserConfirmedPoolWrapperContract] =
+    useState<PoolWrapperAllowanceState>({});
+
+  useEffect(() => {
+    setHasUserConfirmedPoolWrapperContract(confirmed => {
+      return {
+        ...confirmed,
+        [networkInstance]: {
+          [poxContracts['Pox4']]: true,
+          [poxContracts['WrapperFastPool']]:
+            getAllowanceContractCallersFastPoolQuery?.data?.type === ClarityType.OptionalSome,
+          [poxContracts['WrapperRestake']]:
+            getAllowanceContractCallersRestakeQuery?.data?.type === ClarityType.OptionalSome,
+          [poxContracts['WrapperOneCycle']]:
+            getAllowanceContractCallersOneCycleQuery?.data?.type === ClarityType.OptionalSome,
+        },
+      };
+    });
+  }, [
+    poxContracts,
+    networkInstance,
+    getAllowanceContractCallersFastPoolQuery?.data?.type,
+    getAllowanceContractCallersRestakeQuery?.data?.type,
+    getAllowanceContractCallersOneCycleQuery?.data?.type,
+    setHasUserConfirmedPoolWrapperContract,
+  ]);
```

Comment:
Setting state like this from the render process is very side effecty. How else might we do it?

Reply:
refactored a bit